addons/mass_mailing_marketing_card/models/card_card.py

```diff
@@ -0,0 +1,42 @@
+from collections import defaultdict
+
+from odoo import models, osv
+
+
+class CardCard(models.Model):
+    _inherit = 'card.card'
+
+    def _create_if_not_exists(self, card_values_all):
+        """Create new cards based on precommit values."""
+        if not card_values_all:
+            return
+
+        # only keep the last one
+        values_from_campaign_res_pair = dict()
+        for values in card_values_all:
+            values_from_campaign_res_pair[(values['campaign_id'], values['res_id'])] = values
```

Comment:
Correct me if I'm wrong here, but it seems like a caller has a list of tuple, transforms it as a dict, sends it here, to then re-transform it as a list of tuple... ?